datahub/dnb_api/test/test_utils.py

```diff
@@ -423,3 +425,130 @@ def test_post_dnb_data_invalid(
+                {'key': 'value'},
+            ],
+        }
+        mocker = requests_mock.get(
+            DNB_UPDATES_URL,
+            status_code=200,
+            json=expected_response,
+        )
+        response = get_company_update_page(last_updated_after, cursor)
+
+        assert mocker.last_request.qs.get('last_updated_after') == [last_updated_after]
+        assert mocker.last_request.qs.get('cursor') == [cursor or '']
+
+        assert response == expected_response
+
+    @pytest.mark.parametrize(
+        'dnb_response_status',
+        (
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+            status.HTTP_405_METHOD_NOT_ALLOWED,
+            status.HTTP_500_INTERNAL_SERVER_ERROR,
+        ),
+    )
+    def test_dnb_service_error(
+        self,
+        caplog,
+        requests_mock,
+        dnb_response_status,
+    ):
+        """
+        Test if the dnb-service returns a status code that is not
+        200, we log it and raise the exception with an appropriate
+        message.
+        """
+        requests_mock.get(
+            DNB_UPDATES_URL,
+            status_code=dnb_response_status,
+        )
+
+        with pytest.raises(DNBServiceError) as e:
+            get_company_update_page(last_updated_after='foo')
+
+        expected_message = f'DNB service returned an error status: {dnb_response_status}'
+
+        assert e.value.args[0] == expected_message
+        assert len(caplog.records) == 1
+        assert caplog.records[0].getMessage() == expected_message
+
+    @pytest.mark.parametrize(
+        'request_exception, expected_exception, expected_message',
+        (
+            (
+                ConnectionError,
+                DNBServiceConnectionError,
+                'Encountered an error connecting to DNB service',
+            ),
+            (
+                ConnectTimeout,
+                DNBServiceConnectionError,
+                'Encountered an error connecting to DNB service',
+            ),
+            (
+                Timeout,
+                DNBServiceTimeoutError,
+                'Encountered a timeout interacting with DNB service',
+            ),
+            (
+                ReadTimeout,
+                DNBServiceTimeoutError,
+                'Encountered a timeout interacting with DNB service',
+            ),
+        ),
+    )
+    def test_get_company_dnb_service_request_error(
+        self,
+        caplog,
+        requests_mock,
+        request_exception,
+        expected_exception,
+        expected_message,
+    ):
+        """
+        Test if there is an error connecting to dnb-service, we log it and raise
+        the exception with an appropriate message.
+        """
+        requests_mock.get(
+            DNB_UPDATES_URL,
+            exc=request_exception,
+        )
+
+        with pytest.raises(expected_exception) as e:
```

Comment:
This one is usually called excinfo:
```suggestion
        with pytest.raises(expected_exception) as excinfo:
```